### PR TITLE
feat[CWPP-510] Enable unit consumption per ECS container

### PR DESCRIPTION
**Pull request by shankarPlerion**
Summary:
Currently, we were charging customer per ECS task definition. We now want to bill customer per AWS::ECS::TaskDefinition ContainerDefinition as we each of containers in a TaskDefinition.

In this PR, we acknowledge count of containers in AWS::ECS::TaskDefinition.

**Developer testing**

Testing in `us-west-2` region only that had 1 ECS service, 1 Cluster and 1 task definition(with 2 containers definition). 
 = > `node index.js -p AWS -s ECS`
`AWS_MAPPING` looks like:
```json
{
  "total": 4,
  "ECS": {
    "AWS::ECS::Cluster": 1,
    "AWS::ECS::Service": 1,
    "AWS::ECS::TaskDefinition": 2,
  },
}
```

and `AWS-output.json` file looks like:
```json
{
  "TOTAL": 4,
  "CSPM_UNITS": 4,
  "CWPP_UNITS": 20,
  "TOTAL_UNITS": 24,
  "DAYS_PER_MONTH": 30,
  "CSPM_UNITS_PER_MONTH": 120,
  "CWPP_UNITS_PER_MONTH": 600,
  "TOTAL_UNITS_PER_MONTH": 720,
  "INFO": {
    "CSPM_CIEM_ASSET_UNIT_BASE_CONSUMPTION": 1,
    "CWPP_ASSET_UNIT_BASE_CONSUMPTION": 10,
    "TOTAL_DAYS_PER_MONTH": 30
  },
  "DAILY": {
    "CSPM_CIEM_ASSET_UNITS": 4,
    "CWPP_ASSET_UNITS": 20,
    "TOTAL_ASSET_UNITS": 24
  },
  "MONTHLY": {
    "CSPM_CIEM_ASSET_UNITS": 120,
    "CWPP_ASSET_UNITS": 600,
    "TOTAL_ASSET_UNITS": 720
  }
}
```

### CODE REVIEW

File: AWS/aws.js [hunk 1]
```diff
@@ -557,10 +558,21 @@ const queryECSService = async (serviceName, resourceType, region) => {
...
-          serviceName,
-          ECS_TASK_DEFINITION,
-          taskDefinitionArns.length
+
+        await Promise.all(
+          [...new Set(taskDefinitionArns)].map(async (taskDefinition) => {
+            let getTaskDefCmd = new DescribeTaskDefinitionCommand({
+              taskDefinition
+            });
+            let getTaskDefRes = await client?.send(getTaskDefCmd);
```
Comment (sam-plerion) [inline AWS/aws.js:573]: ❓ What's the expected range for the length of `taskDefinitionArns`? Uncontrolled concurrency could result in throttling on high numbers.

Comment (shankarPlerion) [reply AWS/aws.js:573]: Max length of `taskDefinitionArns` cannot be more than 10. Logic to break into chunk was already handled.

Comment (alborz-plerionaut) [reply AWS/aws.js:573]: @shankarPlerion @plerionaut task-def per service: **5000**, container per task-def: **10**
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-quotas.html

Comment (shankarPlerion) [reply AWS/aws.js:573]: @alborz-plerionaut I think the quota you have referenced is for provisioning resource, not for getting details.

Comment (alborz-plerionaut) [reply AWS/aws.js:573]: @shankarPlerion Ah I thought it was related to our discussion, never mind then

**Review by keshav-plerion — APPROVED** [review]
LGTM!

**Pull request merged**